### PR TITLE
feat: upgrade docusaurus version 2

### PR DESCRIPTION
**Pull request by KishaniKandasamy**
Changes:
upgrade docusaurus version 2

Screenshots of the change:

Screenshot (52)

### CODE REVIEW

**Comment by KishaniKandasamy** [comment]
Hi Zhiyuan,
As the  manual migration process should be run after the automated migration process, to debug issues in the migration CLI output,
I did both the automated cli and  manual migration Here.
I have tested on  both linux and windows OS that the  \\\  won't be an issue in  automatically generated files while running  automated migration CLI ):

Thanks.
Screenshot (53)

**Comment by juzhiyuan** [comment]
cc @guoqqqi to review first

File: website/blog/2020-08-22-new-website.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ---
 title: New website for Apache APISIX
```
Comment (guoqqqi) [inline website/blog/2020-08-22-new-website.md:2]: The title of the document is best use "..." , like 
`title: "New website for Apache APISIX"`

File: website/blog/2021-01-21-run-ingress-apisix-on-amazon-eks.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "Run Ingress APISIX on Amazon EKS"
+title: Run Ingress APISIX on Amazon EKS
```
Comment (guoqqqi) [inline website/blog/2021-01-21-run-ingress-apisix-on-amazon-eks.md:2]: `" "`

File: website/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+          "to": "docs/team",
+          "label": "Team",
+          "position": "left"
+        },
+        {
+          "to": "/help",
+          "label": "Help",
+          "position": "left"
+        }
+      ]
```
Comment (guoqqqi) [inline website/docusaurus.config.js:105]: nav should be right in the v1 version

**Comment by guoqqqi** [comment]
It looks great overall, but there is a problem. I'm sorry, I didn't find an update about the footer. I'm not sure if the default footer needs to be changed. Can you provide relevant images? Maybe we need to use：
image

Thanks

**Comment by KishaniKandasamy** [comment]
> It looks great overall, but there is a problem. I'm sorry, I didn't find an update about the footer. I'm not sure if the default footer needs to be changed. Can you provide relevant images? Maybe we need to use：
> image
> 
> Thanks

Yes  When I click on  **swizzle** it says like this.
Screenshot (54)

So I update the footer in docusaurus.config.js with themeConfig.footer as  mentioned in the guide.
Screenshot (56)

**You could find the update  for footer here,**
Screenshot (55)

File: website/blog/2020-08-22-new-website.md (hunk 1 above)
Comment (KishaniKandasamy) [reply website/blog/2020-08-22-new-website.md:2]: Yes.This was  happened  while doing automated migration. As **it is compatible for both the OS** I didn't change it .
I will change it  :)

File: website/docusaurus.config.js (hunk 3 above)
Comment (juzhiyuan) [reply website/docusaurus.config.js:105]: It doesn't have to be IMHO, keep the original V2 style is ok.

File: website/blog/2020-08-22-new-website.md (hunk 1 above)
Comment (juzhiyuan) [reply website/blog/2020-08-22-new-website.md:2]: @guoqqqi Please be careful with using `is best`, we'd better use something, not best or must.

Comment (juzhiyuan) [reply website/blog/2020-08-22-new-website.md:2]: Actually, `title: New website for Apache APISIX` is ok to use 🤔

File: website/docusaurus.config.js (hunk 3 above)
Comment (guoqqqi) [reply website/docusaurus.config.js:105]: Sorry, it’s my fault, I didn’t make it clear, I actually want to ask if we need to be consistent with V1.

File: website/blog/2020-08-22-new-website.md (hunk 1 above)
Comment (guoqqqi) [reply website/blog/2020-08-22-new-website.md:2]: Okay, thank you for your suggestions, I am still trying to learn from them, sorry, I didn't mean to do this.

**Comment by guoqqqi** [comment]
> > It looks great overall, but there is a problem. I'm sorry, I didn't find an update about the footer. I'm not sure if the default footer needs to be changed. Can you provide relevant images? Maybe we need to use：
> > image
> > Thanks
> 
> Yes When I click on **swizzle** it says like this.
> Screenshot (54)
> 
> So I update the footer in docusaurus.config.js with themeConfig.footer as mentioned in the guide.
> Screenshot (56)
> 
> **You could find the update for footer here,**
> Screenshot (55)

Thanks, I found it, look good to me
image

File: website/docusaurus.config.js (hunk 3 above)
Comment (juzhiyuan) [reply website/docusaurus.config.js:105]: no no no, don't have to haha

**Comment by juzhiyuan** [comment]
Fine, then I will involve reviewing this PR sooner or later.

**Comment by imjoey** [comment]
@KishaniKandasamy nice job and many thanks to you.

May I ask that if we could also add `Editing This Page` button for every single page? You could refer to the `editUrl` config item within https://v2.docusaurus.io/docs/migration/manual/#updated-fields .  I guess people could be significantly encourage d to contribute to this repo.

@juzhiyuan any thoughts? Thanks.

**Comment by juzhiyuan** [comment]
@imjoey Of course!

Currently, we need to finish the migration haha, could you please help to review when you are free?

**Comment by imjoey** [comment]
> @imjoey Of course!
> 
> Currently, we need to finish the migration haha, could you please help to review when you are free?

@juzhiyuan sure, let me take this review.

File: website/static/css/custom.css [hunk 4]
```diff
@@ -1,12 +1,20 @@
-/**
- * Copyright (c) Facebook, Inc. and its affiliates.
+/*
```
Comment (imjoey) [inline website/static/css/custom.css]: @KishaniKandasamy We'd better to keep the original MIT License announcement here.

File: website/blog/2020-08-22-new-website.md (hunk 1 above)
Comment (imjoey) [reply website/blog/2020-08-22-new-website.md:2]: @KishaniKandasamy @guoqqqi I would prefer to leave the double quote out of the title, as well as other fields, which is the commonly used way in docusaurus. Thanks.

**Review by imjoey — CHANGES_REQUESTED** [review]
@KishaniKandasamy we probably need to finish another several modifications:

- By referring to https://v2.docusaurus.io/docs/migration#docusaurus-2-structure , we should move `/docs` directory under `/website`, as `/website/docs`. Then the command `COPY ./docs /app/docs` in the Dockerfile could be safely removed.
- In v2, the build artifacts have been relocated to `website/build`, instead of `website/build/apache-apisix`, pls see https://v2.docusaurus.io/docs/migration/manual/#update-references-to-the-build-directory for details. So the commands in `.github/workflows/deploy.yml` are also needed for update.

Thanks.

File: website/static/css/custom.css (hunk 4 above)
Comment (KishaniKandasamy) [reply website/static/css/custom.css]: Fine then I will remove the ASF header and leave it as it is 👍 .

Comment (juzhiyuan) [reply website/static/css/custom.css]: @KishaniKandasamy 👏

**Comment by imjoey** [comment]
> @KishaniKandasamy we probably need to finish another several modifications:
> 
> * By referring to https://v2.docusaurus.io/docs/migration#docusaurus-2-structure , we should move `/docs` directory under `/website`, as `/website/docs`. 

Hi @KishaniKandasamy , you could only do this before we merge this PR.


The following jobs can be done in other PRs. Thanks.
> ... Then the command `COPY ./docs /app/docs` in the Dockerfile could be safely removed.
> * In v2, the build artifacts have been relocated to `website/build`, instead of `website/build/apache-apisix`, pls see https://v2.docusaurus.io/docs/migration/manual/#update-references-to-the-build-directory for details. So the commands in `.github/workflows/deploy.yml` are also needed for update.
> 
> Thanks.

**Comment by KishaniKandasamy** [comment]
> > @KishaniKandasamy we probably need to finish another several modifications:
> > 
> > * By referring to https://v2.docusaurus.io/docs/migration#docusaurus-2-structure , we should move `/docs` directory under `/website`, as `/website/docs`.
> 
> Hi @KishaniKandasamy , you could only do this before we merge this PR.
> 
> The following jobs can be done in other PRs. Thanks.
> 
> > ... Then the command `COPY ./docs /app/docs` in the Dockerfile could be safely removed.
> > 
> > * In v2, the build artifacts have been relocated to `website/build`, instead of `website/build/apache-apisix`, pls see https://v2.docusaurus.io/docs/migration/manual/#update-references-to-the-build-directory for details. So the commands in `.github/workflows/deploy.yml` are also needed for update.
> > 
> > Thanks.

Hi Imjoey,
I didn't move /docs as it says something like this.
Screenshot (59)



Now  I have done some modifications that  you mentioned here.
1.Add editUrl(It will work fine once merged this PR)
2.Remove ASF header from custom.css
3.Move /docs under website directory 

I 've pushed these  changes with the last two commits.

Thanks

**Comment by imjoey** [comment]
> Now I have done some modifications that you mentioned here.
> 1.Add editUrl(It will work fine once merged this PR)
> 2.Remove ASF header from custom.css
> 3.Move /docs under website directory
> 
> I 've pushed these changes with the last two commits.
> 
> Thanks

@KishaniKandasamy nice job and many thanks.

**Review by imjoey — APPROVED** [review]
/lgtm

thank you @KishaniKandasamy .

**Comment by imjoey** [comment]
ping @juzhiyuan for reviewing at your convenience. Thanks.

**Comment by KishaniKandasamy** [comment]
> /lgtm
> 
> thank you @KishaniKandasamy .

It's my pleasure ;)

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**